catalog/rocks/staging_test.go

```diff
@@ -0,0 +1,230 @@
+	}
+	if len(res) != numOfEntries {
+		t.Errorf("got unexpected number of results. expected=%d, got=%d", numOfEntries, len(res))
+	}
+	for i, e := range res {
+		if e.Path != Path(fmt.Sprintf("entry%04d", i)) {
+			t.Fatalf("got unexpected entry from list at index %d: expected entry%04d, got: %s", i, i, e.Path)
+		}
+	}
+}
+
+func TestSeek(t *testing.T) {
+	s := prepareMgr(t)
+	numOfEntries := 100
+	for i := 0; i < numOfEntries; i++ {
+		err := s.SetEntry(context.Background(), "t1", Path(fmt.Sprintf("entry%04d", i)), entry("addr1"))
+		if err != nil {
+			t.Fatalf("got unexpected error: %v", err)
+		}
+	}
+	it, _ := s.ListEntries(context.Background(), "t1")
+	if !it.SeekGE("entry0050") {
+		t.Fatal("iterator seek expected to return true, got false")
+	}
+	if it.Value().Path != "entry0050" {
+		t.Fatalf("got unexpected entry after iterator seek. expected=entry0050, got=%s", it.Value().Path)
+	}
+	if !it.Next() {
+		t.Fatal("iterator next expected to return true, got false")
+	}
+	if it.Value().Path != "entry0051" {
+		t.Fatalf("got unexpected entry after iterator seek. expected=entry0051, got=%s", it.Value().Path)
+	}
+	if it.SeekGE("entry1000") {
+		t.Fatal("iterator seek expected to return false, got true")
+	}
+	if !it.SeekGE("entry0060a") {
+		t.Fatal("iterator seek expected to return true, got false")
+	}
+	if it.Value().Path != "entry0061" {
+		t.Fatalf("got unexpected entry after iterator seek. expected=entry0061, got=%s", it.Value().Path)
+	}
+	if !it.Next() {
+		t.Fatal("iterator next expected to return true, got false")
+	}
+}
+
+func TestDeleteAndTombstone(t *testing.T) {
+	s := prepareMgr(t)
+	_, err := s.GetEntry(context.Background(), "t1", "entry1")
+	if !errors.Is(err, db.ErrNotFound) {
+		t.Fatalf("error different than expected. expected=%v, got=%v", db.ErrNotFound, err)
+	}
+	err = s.SetEntry(context.Background(), "t1", "entry1", nil)
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	e, err := s.GetEntry(context.Background(), "t1", "entry1")
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	if e != nil {
+		t.Fatalf("expected nil but got entry: %v", e)
+	}
+	err = s.SetEntry(context.Background(), "t1", "entry1", entry("addr3"))
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	e, err = s.GetEntry(context.Background(), "t1", "entry1")
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	if e.Address != "addr3" {
+		t.Fatalf("got unexpected entry address. expected=%s, got=%s", "addr3", e.Address)
+	}
+	err = s.DeleteEntry(context.Background(), "t1", "entry1")
+	if err != nil {
+		t.Fatalf("got unexpected error: %v", err)
+	}
+	_, err = s.GetEntry(context.Background(), "t1", "entry1")
+	if !errors.Is(err, db.ErrNotFound) {
+		t.Fatalf("error different than expected. expected=%v, got=%v", db.ErrNotFound, err)
+	}
+}
+
+func entry(addr string) *Entry {
+	return &Entry{
+		LastModified: time.Now(),
+		Address:      addr,
+		Metadata:     entryMetadata,
```

Comment:
1. can be nil
2. make a new one for each entry, so they will not share the same reference